src/libltfs/tape.c

```diff
@@ -3570,3 +3570,48 @@ int tape_set_profiler(struct device_data *dev, char *work_dir, bool enable)
 
 	return dev->backend->set_profiler(dev->backend_data, work_dir, enable);
 }
+
+/**
+ * Prepares GRAO parameter list, and sends rao request.
+ * All data should be set in rao before this is called.
+ * @param dev a pointer to the tape device
+ * @param rao a pointer to the rao data
+ * @param [out] ret_buf the returned buffer from grao.
+ * @return 0 on success or a negative value on error
+ */
+int tape_rao_request(struct device_data *dev, struct rao_mod *rao)
+{
+	int ret = 0;
+
+	/* check file size */
+	if (rao->num_of_files <= 0) {
+		/* rao list will be cleared if call size is less than zero */
+		rao->in_buf = NULL;
+		ltfsmsg(LTFS_DEBUG, 17277D, "Clear Called");
+	}
+
+	if (rao->in_buf == NULL){
+		return EDEV_INTERNAL_ERROR;
+	}
+
+	/* run GRAO */
+	ret = dev->backend->grao(dev->backend_data, (const unsigned char *)rao->in_buf, rao->num_of_files);
+	if (ret < 0) {
+		ltfsmsg(LTFS_ERR, 17278E, "GRAO", ret); //GRAO command returns error
+		return ret;
+	}
+	else if (rao->num_of_files <= 0) {
```

Comment:
Please do not make a line break between `}` and `else`.

Reply:
modified in https://github.com/LinearTapeFileSystem/ltfs/pull/288/commits/87b5a2d44974149001f8f29ae12a3718b102e69f